### PR TITLE
Rename local-gadget to ig

### PR DESCRIPTION
**Pull request by burak-ok**
# Rename local-gadget to ig

This PR renames every occurrence of `local-gadget` to `ig`.
Be it in the code, a comment, Github Workflow, ...

I tried to separate the commits as cleanly as possible

Test for the release workflow change: https://github.com/burak-ok/inspektor-gadget/releases/tag/v0.0.1

## Discussion/Question
Having `ig` in the code could be confusing since the whole project is called `Inspektor Gadget` -> `IG`
Should we pick another name for inside the code? Maybe currently only rename the executable so we have more think about the naming in the code?

### CODE REVIEW

File: .github/workflows/inspektor-gadget.yml [hunk 1]
```diff
@@ -862,20 +862,26 @@ jobs:
...
         done
-    - name: Upload Gadget Release *.tar.gz
+    - name: Upload kubectl-gadget Release *.tar.gz
       uses: csexton/release-asset-action@v2
       with:
-        pattern: "*-gadget-*-*-tar-gz/*-gadget-*-*.tar.gz"
+        pattern: "kubectl-gadget-*-*-tar-gz/kubectl-gadget-*-*.tar.gz"
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        release-url: ${{ steps.create_release.outputs.upload_url }}
+    - name: Upload IG Release *.tar.gz
```
Comment (mauriciovasquezbernal) [inline .github/workflows/inspektor-gadget.yml:881]: Upload IG Release ->  Upload IG Binary?

File: .github/workflows/inspektor-gadget.yml [hunk 2]
```diff
@@ -862,20 +862,26 @@ jobs:
...
     - name: Compute checksums for all artifacts
       shell: bash
       run: |
-        for i in *-gadget-*-*-tar-gz/*-gadget-*-*.tar.gz inspektor-gadget-${{ github.ref_name }}.yaml; do
+        for i in kubectl-gadget-*-*-tar-gz/kubectl-gadget-*-*.tar.gz ig-*-*-tar-gz/ig-*-*.tar.gz inspektor-gadget-${{ github.ref_name }}.yaml; do
           hash=$(sha256sum $i | cut -d' ' -f1)
           echo "${hash}  $(basename $i)" >> inspektor-gadget-${{ github.ref_name }}_checksums.txt
         done
-    - name: Upload Gadget Release *.tar.gz
+    - name: Upload kubectl-gadget Release *.tar.gz
```
Comment (mauriciovasquezbernal) [inline .github/workflows/inspektor-gadget.yml:875]: Release -> Binary?

File: pkg/ig-manager/ig-manager.go [hunk 3]
```diff
@@ -26,7 +26,7 @@ import (
 	tracercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/tracer-collection"
 )
 
-type LocalGadgetManager struct {
+type IgManager struct {
```
Comment (mauriciovasquezbernal) [inline pkg/ig-manager/ig-manager.go:29]: This is the only one I'm not totally sure about. However I think for the time being we can use `IGManager` and then we can look for a better name. (once the reworking of kubectl-gadget being done by @flyth is finished).

**Review by mauriciovasquezbernal — APPROVED** [review]
IMO we can merge as it's. We can always rename those internal types later on if needed.

File: docs/ci.md [hunk 4]
```diff
@@ -7,7 +7,7 @@ description: >
 
 This repository uses GitHub Actions to create a custom Continuous Integration
 (CI) workflow named `Inspektor Gadget CI`. This workflow compiles and uploads
-the Inspektor Gadget CLI, local-gadget and gadget container image and runs their
+the Inspektor Gadget CLI, IG and gadget container image and runs their
```
Comment (blanquicet) [inline docs/ci.md:10]:
```suggestion
the Inspektor Gadget CLIs: `kubectl-gadget` and `ig`; and gadget container image and runs their
```

File: docs/devel/CONTRIBUTING.md [hunk 5]
```diff
@@ -124,11 +124,11 @@ $ export KUBECONFIG=... # not needed if valid config in $HOME/.kube/config
 $ make integration-tests
 ```
 
-### Integration tests for Local Gadget
+### Integration tests for IG
 
 #### Kubernetes
 
-The integration tests for local gadget uses minikube for testing different container runtimes.
+The integration tests for ig uses minikube for testing different container runtimes.
```
Comment (blanquicet) [inline docs/devel/CONTRIBUTING.md:131]:
```suggestion
The integration tests for `ig` uses minikube for testing different container runtimes.
```

File: docs/gadgets/advise/network-policy.md [hunk 6]
```diff
@@ -169,6 +169,6 @@ namespace "demo" deleted
 
 - When using the Docker bridge as CNI, pod-to-pod source IP is lost with services. This generates wrong ingress policies. https://github.com/kubernetes/minikube/issues/11211
 
-### With local-gadget
+### With IG
 
-This gadget is specific to Kubernetes and can't be used with local-gadget.
+This gadget is specific to Kubernetes and can't be used With IG.
```
Comment (blanquicet) [inline docs/gadgets/advise/network-policy.md:174]:
```suggestion
This gadget is specific to Kubernetes and can't be used with `ig`.
```

File: docs/devel/CONTRIBUTING.md [hunk 7]
```diff
@@ -84,14 +84,14 @@ go: downloading github.com/giantswarm/crd-docs-generator v0.7.1
 Wrote /work/pkg/gadgettracermanager/containers-map/containersmap_bpfel.go
 ```
 
-### Building Local Gadget
+### Building IG
```
Comment (blanquicet) [inline docs/devel/CONTRIBUTING.md:87]: nit: I'm not sure if it should be "Building `ig`" and the same for all the "With `ig`". What do other folks think?

File: docs/gadgets/profile/cpu.md [hunk 8]
```diff
@@ -135,24 +135,24 @@ Finally, we need to clean up our pod:
...
 ```bash
 $ docker run -d --rm --name random busybox cat /dev/urandom > /dev/null
 ```
 
-* Start local-gadget:
+* Start `ig`:
 
 ```bash
-$ sudo ./local-gadget profile cpu -K --containername random --runtimes docker
+$ sudo ./ig profile cpu -K --containername random --runtimes docker
```
Comment (blanquicet) [inline docs/gadgets/profile/cpu.md:149]: For `kubectl-gadget` we are using `kubectl gadget` (so assuming users has it installed in the host). Why we aren't doing the same for `ig`?. If we all agree, it should be done in another PR as it is completed unrelated:

```suggestion
$ sudo ig profile cpu -K --containername random --runtimes docker
```

File: docs/gadgets/trace/capabilities.md [hunk 9]
```diff
@@ -270,12 +270,12 @@ namespaces owning the mount namespace, that is the parent user namespace:
 if (!ns_capable(mnt_ns->user_ns, CAP_SYS_ADMIN) || ...
 ```
 
-### With local-gadget
+### With IG
 
-Start local-gadget:
+Start `IG`:
```
Comment (blanquicet) [inline docs/gadgets/trace/capabilities.md:275]:
```suggestion
Start `ig`:
```

File: docs/gadgets/trace/capabilities.md [hunk 10]
```diff
@@ -270,12 +270,12 @@ namespaces owning the mount namespace, that is the parent user namespace:
...
 
 ```bash
-$ local-gadget trace capabilities -r docker -c test
+$ ig trace capabilities -r docker -c test
```
Comment (blanquicet) [inline docs/gadgets/trace/capabilities.md:278]: Not related with this PR though:

```suggestion
$ sudo ig trace capabilities -r docker -c test
```

File: docs/ig.md [hunk 11]
```diff
@@ -1,25 +1,25 @@
 ---
-title: local-gadget
+title: IG
 weight: 80
 description: >
-  Description of the local-gadget tool.
+  Description of the IG tool.
```
Comment (blanquicet) [inline docs/ig.md:5]: The same here. I'm wondering if we should always use lowercase `ig` because we sometimes use IG to refer to the one for Kubernetes or the whole project.

cc @mauriciovasquezbernal 

```suggestion
  Description of the ig tool.
```

File: pkg/ig-manager/ig-manager.go (hunk 3 above)
Comment (blanquicet) [reply pkg/ig-manager/ig-manager.go:29]: Yeah, It is strange. I agree we can go ahead with `IGManager` (and `igManager` as a variable) and then look for a better one after the refactoring is finished.

File: docs/ig.md (hunk 11 above)
Comment (mauriciovasquezbernal) [reply docs/ig.md:5]: Yes, I think we should call it with lowercase everywhere to make sure it refers to the CLI tool.

File: docs/gadgets/profile/cpu.md (hunk 8 above)
Comment (burak-ok) [reply docs/gadgets/profile/cpu.md:149]: Sometimes we also use `ig profile....` instead of `./ig profile....` in other doc files. We should consolidate the whole documentation (An Issue already exists i think)

File: docs/devel/CONTRIBUTING.md (hunk 7 above)
Comment (burak-ok) [reply docs/devel/CONTRIBUTING.md]: Removed every uppercase `IG`, except one in `cmd/kubectl-gadget/undeploy.go:167`. There it doesn't refer to `local-gadget`
```go
// Let's try to remove components of IG versions before v0.5.0,
// just in case somebody has a newer CLI but is trying to remove
// an old version of Inspektor Gadget from the cluster. Given
// that this is a best effort work, we don't track any error.

// kube-system/gadget daemon set
k8sClient.AppsV1().DaemonSets("kube-system").Delete(
context.TODO(), "gadget", metav1.DeleteOptions{},
)
```

File: docs/install.md [hunk 12]
```diff
@@ -168,9 +168,9 @@ possible that different versions work well together, we don't provide
 any guarantee in those cases. We'll visit this policy again once we
 approach to the v1.0 release.
 
-## Installing local-gadget
+## Installing IG
```
Comment (blanquicet) [inline docs/install.md:171]: dito

**Review by blanquicet — APPROVED** [review]
LGTM!

File: README.md [hunk 13]
```diff
@@ -198,10 +198,10 @@ programs are and how Inspektor Gadget uses them is briefly explained here:
 
 You can read further details about the architecture [here](docs/architecture.md).
 
-## local-gadget
+## IG
```
[the remark this answers is not in the document]
Comment (burak-ok) [reply README.md:201]: Thought that i had all uppercase `IG`... Apparently my regex was too restrictive with `IG` on the end of a line :D
